### PR TITLE
Bug 1697462, update node dependencies

### PR DESCRIPTION
**Pull request by Pike**
* Update babel and eslint

Drafting this PR for now, as I expect it to conflict with the update-flow PR on package-lock.json.

And once that's done, I'll add more commits for react-scripts and caniuse and such.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -90,8 +90,8 @@ format:
 	make pyupgrade
 	make black
 
-lint-frontend:
-	"${DC}" run --rm -w //app/frontend webapp ./node_modules/.bin/eslint src/
+eslint:
```
Comment (mathjazz) [inline Makefile:93]: Note that we still have `lint-frontend` in other places within the file.

**Review by mathjazz — COMMENTED** [review]
Thanks!

> And once that's done, I'll add more commits for react-scripts and caniuse and such.

We should also update elliptic to the latest version.

File: Makefile [hunk 2]
```diff
@@ -20,7 +20,7 @@ help:
 	@echo "  clean            Forces a rebuild of docker containers"
 	@echo "  shell            Opens a Bash shell in a webapp docker container"
 	@echo "  test             Runs the entire test suite (back and front)"
-	@echo "  test-frontend    Runs the new frontend's test suite (Translate.Next)"
+	@echo "  test-frontend    Runs the new frontend's test suite"
 	@echo "  jest             Runs the jest test runner on all Translate.Next tests"
```
Comment (mathjazz) [inline Makefile:24]: s/Translate.Next/frontend

**Comment by Pike** [comment]
I think this is a good step forward, but there's also still a lot to do:

```
pontoon$ npm outdated
Package                  Current   Wanted         Latest  Location
css-loader                 3.6.0    3.6.0          5.1.3  pontoon
jest                      24.9.0   24.9.0         26.6.3  pontoon
jsdom                    11.12.0  11.12.0         16.5.1  pontoon
mini-css-extract-plugin    0.4.5    0.4.5          1.3.9  pontoon
prettier                   2.2.0    2.2.0          2.2.1  pontoon
react                    16.14.0  16.14.0         17.0.1  pontoon
react-dom                16.14.0  16.14.0         17.0.1  pontoon
react-table               6.11.5   6.11.5          7.6.3  pontoon
react-test-renderer      16.14.0  16.14.0         17.0.1  pontoon
style-loader              0.19.1   0.19.1          2.0.0  pontoon
webpack                   4.46.0   4.46.0         5.26.3  pontoon
webpack-bundle-tracker     0.4.3    0.4.3  1.0.0-alpha.1  pontoon
webpack-cli               3.3.12   3.3.12          4.5.0  pontoon
```

```
pontoon/frontend$ yarn outdated
yarn outdated v1.22.10
info Color legend :
 "<red>"    : Major Update backward-incompatible updates
 "<yellow>" : Minor Update backward-compatible features
 "<green>"  : Patch Update backward-compatible bug fixes
Package                   Current Wanted  Latest      Package Type URL

abortcontroller-polyfill  1.3.0   1.3.0   1.7.1       dependencies https://github.com/mo/abortcontroller-polyfill#readme

connected-react-router    6.8.0   6.8.0   6.9.1       dependencies https://github.com/supasate/connected-react-router#readme
diff-match-patch          1.0.4   1.0.4   1.0.5       dependencies https://github.com/JackuB/diff-match-patch#readme

highcharts                7.2.2   7.2.2   9.0.1       dependencies http://www.highcharts.com

highcharts-react-official 2.2.2   2.2.2   3.0.0       dependencies https://github.com/highcharts/highcharts-react#readme

html-react-parser         0.13.0  0.13.0  1.2.4       dependencies https://github.com/remarkablemark/html-react-parser#readme
intl-pluralrules          1.0.0   1.0.0   1.2.2       dependencies https://github.com/eemeli/intl-pluralrules#readme

jest-each                 24.5.0  24.5.0  26.6.2      dependencies https://github.com/facebook/jest#readme

linkify-it                2.1.0   2.1.0   3.0.2       dependencies https://github.com/markdown-it/linkify-it#readme

react                     16.13.1 16.13.1 17.0.1      dependencies https://reactjs.org/

react-dom                 16.13.1 16.13.1 17.0.1      dependencies https://reactjs.org/

react-linkify             0.2.2   0.2.2   1.0.0-alpha dependencies http://tasti.github.io/react-linkify/

react-redux               7.1.3   7.1.3   7.2.2       dependencies https://github.com/reduxjs/react-redux

react-tabs                3.0.0   3.0.0   3.2.1       dependencies https://github.com/reactjs/react-tabs

react-time-ago            5.0.4   5.0.4   6.2.2       dependencies https://gitlab.com/catamphetamine/react-time-ago#readme
reactour                  1.15.1  1.15.1  1.18.3      dependencies https://github.com/elrumordelaluz/reactour#readme

shortid                   2.2.14  2.2.14  2.2.16      dependencies https://github.com/dylang/shortid#readme

slate                     0.58.4  0.58.4  0.59.0      dependencies https://github.com/ianstormtaylor/slate#readme

slate-react               0.58.4  0.58.4  0.59.0      dependencies https://github.com/ianstormtaylor/slate#readme

styled-components         4.3.2   4.3.2   5.2.1       dependencies https://styled-components.com/
```

**Review by mathjazz — APPROVED** [review]
Changes look good and make a bunch of things much cleaner! Nice job!

I tested locally and on stage and didn't spot any issues.

**Pull request merged**